test/python/quantum_info/operators/symplectic/test_sparse_pauli_op.py

```diff
@@ -273,79 +356,101 @@ class TestSparsePauliOpMethods(QiskitTestCase):
 
     RNG = np.random.default_rng(1994)
 
-    def random_spp_op(self, num_qubits, num_terms):
+    def setUp(self):
+        super().setUp()
+
+        def bind_one(a):
+            parameters = a.parameters
+            return complex(a.bind(dict(zip(parameters, [1] * len(parameters)))))
+
+        self.vbind_one = np.vectorize(bind_one, otypes=[complex])
```

Comment:
This is rather difficult to read for a testing function, and it's a bit mysterious what its purpose is.  I'm guessing the intent is to remove all the `Parameter` components of a `ParameterExpression`, and end up left with just the pre-multiplier on the coefficient that comes from the phase factor on the Pauli?  It would help to have a comment.

The `setUp` method is called before every test, so the `np.vectorize` call isn't being cached here, if that was your intent.  You could define this outside the class as a regular method, if you want to keep that.

Reply:
The intent is that the operation will produce a numerical error, but not the same when compared with the `Parameter` remaining, so an error will result. Therefore, I bind the some value and then compare. I'll add some comments about this testing methods.

The intent is not caching, this line is not slow.
<img width="533" alt="image" src="https://user-images.githubusercontent.com/6814928/192658859-298d4e8a-5f0e-4b9e-8d07-7fee9e989aa5.png">

Reply:
In that case, it may be cleaner just to have a regular function
```python
def bind_parameters_to_one(array):
    def bind_one(expression):
       parameters = expression.parameters
       return complex(...)
    return np.vectorize(bind_one, otypes=[complex])(array)
```
or even a static method:
```python
class TestCase:
    def _bind_one(expression):
        return complex(...)
    bind_parameters_to_one = staticmethod(np.vectorize(_bind_one, otypes=[complex])
```
(basically it's just a bit odd to me that the function binding doesn't use one of the standard Python ways of binding to the class, but instead does it to the instance dict.)

Reply:
It is certainly strange to create something that is static for each instance. Thank you for your suggestion.